### DISCUSSION TITLE
Building a LibreChat structured Agent for MS Sharepoint using Graph APIs

### DISCUSSION THREAD

**Discussion by aleibovici** [danny-avila/LibreChat#4491] [Feature Requests & Suggestions]
Hi there, I want to build a structured agent for SharePoint using Graph APIs to retrieve data and metadata from files stored in SharePoint sites, OneDrive, etc. Before I get started, has anyone built something similar?

**Comment by jmaddington** [comment]
@aleibovici did you start on this?

**Reply by aleibovici** [reply]
@jmaddington No, I haven't started on that yet. If you decide to start, I would be keen to contribute. Otherwise, I should be able to find some free cycles soon.

**Reply by jmaddington** [reply]
I'd like to know how far you fleshed it out, even if nothing is written down.

How would you handle oauth, for instance, or returning contextual data? Would you parse PDFs and Office documents, or just return that they exist and match a search?

**Reply by jmaddington** [reply]
Someone has already made progress on this: https://github.com/intelequia/LibreChat/tree/sharepoint

**Reply by aleibovici** [reply]
I had a quick look at it, and it seems he made some implementations using Dataverse and Microsoft Graph. However, there is no structured agent created to access SharePoint data and metadata. If I'm correct, I believe @danny-avila  has mentioned that the future of agents is structured. I'd rather create something that will be supported by the new agent framework.

**Comment by DanielJSimons** [comment]
This would be a great feature if implemented..
